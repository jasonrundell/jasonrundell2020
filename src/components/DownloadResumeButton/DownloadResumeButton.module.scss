.root {
  display: inline-block;
  font-size: $sizeSmaller;
  // line-height: $sizeSmall;
  padding: 0.9rem 0.6rem 0.3rem 0.9rem;
  background: $colorPrime;
  border: 1px solid $colorPrimeLight;
  border-radius: $sizeSmallest;
  transition: all, 0.5s, ease;
  animation-delay: 0;
  animation-duration: 1000ms;
  animation-fill-mode: both;
  animation-timing-function: linear;
  animation-iteration-count: 1;

  @media (min-width: $mediumBreakpoint) {
    font-size: $sizeSmall;
  }
  &:link,
  &:visited,
  &:hover,
  &:focus,
  &:active {
    color: $accentColorPrime;
    text-decoration: none;
  }
  &:hover,
  &:focus {
    background: $colorPrimeLight;
    border-color: $accentColorPrime;
  }
  &:active {
    color: $accentColorDark;
    background: $colorPrimeDarker;
  }
  &.active {
    color: $accentColorDark;
  }

  .arrow-icon {
    position: relative;
    top: -1px;
    -webkit-transition: -webkit-transform 0.3s ease;
    transition: -webkit-transform 0.3s ease;
    transition: transform 0.3s ease;
    transition: transform 0.3s ease, -webkit-transform 0.3s ease;
    vertical-align: middle;
    transform: rotate(90deg);
    margin-left: $sizeSmallest;
  }
  .arrow-icon--circle {
    transition: stroke-dashoffset 0.3s ease;
    stroke-dasharray: 95;
    stroke-dashoffset: 95;
  }
  &:hover {
    text-decoration: none;
    .arrow-icon {
      transform: translate3d(5px, 0, 0);
      transform: rotate(90deg);
      stroke: #e9be62;
    }
    .arrow-icon--circle {
      stroke-dashoffset: 0;
      stroke: #e9be62;
    }
    .arrow-icon--arrow {
      stroke: #e9be62;
    }
  }
}
