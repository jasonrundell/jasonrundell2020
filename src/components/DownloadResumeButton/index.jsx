import React from 'react'

// CSS
import {
  root,
  text,
  arrowIcon,
  arrowIconCircle,
  arrowIconArrow,
} from './DownloadResumeButton.module.scss'

import resumePdf from '../../assets/jason-rundell-web-developer-resume.pdf'

const DownloadResumeButton = () => {
  return (
    <div style={{ width: '100%', textAlign: 'right' }}>
      <a href={resumePdf} className={root}>
        <span className={text}>Download résumé</span>
        <svg
          className={arrowIcon}
          xmlns="http://www.w3.org/2000/svg"
          width="32"
          height="32"
          viewBox="0 0 32 32"
        >
          <g
            fill="none"
            stroke="#e9be62"
            strokeWidth="1.5"
            strokeLinejoin="round"
            strokeMiterlimit="10"
          >
            <circle
              className={arrowIconCircle}
              cx="16"
              cy="16"
              r="14.12"
            ></circle>
            <path
              className={arrowIconArrow}
              d="M16.14 9.93L22.21 16l-6.07 6.07M8.23 16h13.98"
            ></path>
          </g>
        </svg>
      </a>
    </div>
  )
}

export default DownloadResumeButton
