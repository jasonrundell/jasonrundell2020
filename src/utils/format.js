/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
