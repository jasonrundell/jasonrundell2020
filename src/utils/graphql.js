export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
