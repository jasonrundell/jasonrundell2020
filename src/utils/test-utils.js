export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
