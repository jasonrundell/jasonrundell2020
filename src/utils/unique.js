const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
