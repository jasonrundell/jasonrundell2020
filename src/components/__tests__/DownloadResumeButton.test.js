import React from 'react'
import { unmountComponentAtNode } from 'react-dom'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import DownloadResumeButton from '../DownloadResumeButton'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('DownloadResumeButton component', () => {
  it('DownloadResumeButton renders correctly', () => {
    const tree = renderer.create(<DownloadResumeButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
