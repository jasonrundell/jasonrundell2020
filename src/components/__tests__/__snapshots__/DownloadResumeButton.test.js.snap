// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadResumeButton component DownloadResumeButton renders correctly 1`] = `
<div
  style={
    Object {
      "textAlign": "right",
      "width": "100%",
    }
  }
>
  <a
    className="root"
    href="/jason-rundell-web-developer-resume.pdf"
  >
    <span
      className="text"
    >
      Download résumé
    </span>
    <svg
      className="arrowIcon"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        stroke="#e9be62"
        strokeLinejoin="round"
        strokeMiterlimit="10"
        strokeWidth="1.5"
      >
        <circle
          className="arrowIconCircle"
          cx="16"
          cy="16"
          r="14.12"
        />
        <path
          className="arrowIconArrow"
          d="M16.14 9.93L22.21 16l-6.07 6.07M8.23 16h13.98"
        />
      </g>
    </svg>
  </a>
</div>
`;
